pkg/crawler/scan.go

```diff
@@ -45,6 +52,57 @@ func SecretsMatch(url, body string, secretsFile *[]string) []scanner.SecretMatch
 	var secrets []scanner.SecretMatched
 
 	if len(*secretsFile) == 0 {
+		ctx, _ := context.WithTimeout(context.Background(), time.Hour*2)
+		sem := semaphore.NewWeighted(int64(runtime.NumCPU()))
+		var wgScanners sync.WaitGroup
+
+		// Buffered channel to control the number of goroutines
+		guard := make(chan struct{}, int64(runtime.NumCPU()))
+
+		// import trufflehog default scanner list
+		for _, trufflehogScanner := range engine.DefaultDetectors() {
+			wgScanners.Add(1)
+
+			// Acquire a slot from the channel
+			guard <- struct{}{}
+
+			go func(s detectors.Detector) {
+				defer func() {
+					// Release the slot back to the channel
+					<-guard
+					wgScanners.Done()
+				}()
+
+				// Check secrets from the scanner against the body of the request
+				res, err := s.FromData(ctx, true, []byte(body))
+				if err != nil {
+					return
+				}
+
+				if len(res) > 0 {
+					for _, resSecret := range res {
+						sec := scanner.Secret{
+							Name:           resSecret.DetectorType.String(),
+							Description:    resSecret.DetectorType.String(),
+							Regex:          "",
+							FalsePositives: []string{},
+							Poc:            "",
+						}
+						secretFound := scanner.SecretMatched{Secret: sec, URL: url, Match: string(resSecret.Raw)}
+
+						// prevent concurent write
+						sem.Acquire(ctx, 1)
+						secrets = append(secrets, secretFound)
+						sem.Release(1)
+					}
+				}
+			}(trufflehogScanner)
+
+		}
+
+		wgScanners.Wait()
+
+		// keep legacy code, the original list contains secrets that are not present in trufflehog like s3 buckets.
```

Comment:
I'm not fully understanding this. I guess now the code performs both trufflehog and cariddi secrets detections right? Maybe we should find a solution for this. 

A simple solution could be move the S3 regexes to `info` of the cariddi engine and run only the trufflehog engine for the secrets. But I'm open to suggestions if you have them.

Also, we have to take in mind that the user can input a file with custom regexes, I don't know if trufflehog can do this. If not, maybe we could leave the custom cariddi engine for the custom secrets.

Let me know your thoughts on this.